server/src/test/java/org/opensearch/index/store/remote/utils/cache/RefCountedCacheTestCase.java

```diff
@@ -138,6 +140,23 @@ public void testPrune() {
         assertEquals(10L, (long) refCountedCache.get("3"));
     }
 
+    public void testPruneWithPredicate() {
+        refCountedCache.put("1", 10L);
+        refCountedCache.decRef("1");
+        refCountedCache.put("2", 10L);
+        refCountedCache.decRef("2");
+        refCountedCache.put("3", 10L);
+
+        Predicate<String> falsePredicate = path -> false;
```

Comment:
Nitpick, but these predicates can probably be inlined instead of defining a local variable

Reply:
Updated.